cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1145,6 +1145,19 @@ class CategoricalBlock : public PandasBlock {
       converted_col =
           std::make_shared<Column>(field(col->name(), out.type()), out.chunked_array());
     } else {
+      // check if all dictionaries are equal
+      const ChunkedArray& data = *col->data().get();
```

Comment:
I put this code in the "else" clause here, because the validation should not be necessary in case the dictionaries where just computed with compute:DictionaryEncode (I assumed, or can that method return a variable dictionary if it is passed a chunked array?)